core/src/main/scala/pink/cozydev/lucille/Parser.scala

```diff
@@ -144,11 +145,9 @@ object Parser {
   def rangeQuery: P[TermRange] = {
     val inclLower = P.charIn('{', '[').map(lowerBound => lowerBound == '[') <* maybeSpace
     val inclUpper = maybeSpace *> P.charIn('}', ']').map(upperBound => upperBound == ']')
-    val boundValue =
-      P.char('*').as(None) | P.not(P.stringIn(reserved)).with1 *> (alpha | digit | P.char(
-        '.'
-      )).rep.string
-        .map(Some(_))
+    val wild = P.char('*').as(None)
+    val txt = (alpha.void | digit.void | P.char('.')).rep.string.map(Some(_))
```

Comment:
What is the `.void` for?

Reply:
oh my gosh, this was annoying to figure out.
What motivates any change at all is new type inference warnings after upgrading to sbt-typelevel 0.5.x

The issue is we were mixing `Parser[Char]`s with `Parser[Unit]` and so the compiler infers `Any` (which triggers the warning) on the `orElse` combinator type param. We add the voids to make everything `Parser[Unit]`.

Reply:
Oh wow!! Good to know!